src/python/zquantum/core/circuit/conversions/pyquil_conversions.py

```diff
@@ -203,3 +221,51 @@ def convert_circuit_to_pyquil(
         program += convert_to_pyquil(gate, program)
 
     return program
+
+
+@singledispatch
+def convert_from_pyquil(obj: Union[pyquil.Program, pyquil.quil.Gate]):
+    pass
+
+
+@convert_from_pyquil.register
+def convert_gate_from_pyquil(gate: pyquil.quil.Gate) -> Gate:
+    number_of_control_modifiers = gate.modifiers.count("CONTROLLED")
+
+    all_qubits = pyquil_qubits_to_numbers(gate.qubits)
+    control_qubits = all_qubits[:number_of_control_modifiers]
+    original_qubits = all_qubits[number_of_control_modifiers:]
```

Comment:
Does "original" here actually mean "target"?

Reply:
Original here refers to qubits that the gate acted on before applying modifiers. I agree that `target_qubits` is a better name for this variable. Fixed in 0ef10bb